otherlibs/dune-rpc/private/dune_rpc_private.ml

```diff
@@ -440,14 +440,25 @@ module Message = struct
     iso (record (both payload message)) to_ from
 end
 
-module Subscribe = struct
-  type t =
-    | Diagnostics
-    | Build_progress
+module Sub = struct
+  type 'a t =
+    { diff : 'a Conv.value
+    ; name : string
+    }
 
-  let sexp =
-    Conv.enum
-      [ ("Diagnostics", Diagnostics); ("Build_progress", Build_progress) ]
+  let progress = { diff = Progress.sexp; name = "progress" }
+
+  let diagnostic =
+    { diff = Conv.list Diagnostic.Event.sexp; name = "diagnostic" }
+
+  let subscribe = Decl.request ~method_:"subscribe" Conv.string Conv.unit
+
+  (* send by the client to cancel a subscription *)
+  let cancel = Decl.notification ~method_:"unsubscribe" Id.sexp
+
+  let sub_update =
```

Comment:
What does this do?

Reply:
This is the notification used under the hood by the server to update streams. It can either provide a new element, or signal that the stream is exhausted.